libmproxy/console/common.py

```diff
@@ -161,6 +165,88 @@ def raw_format_flow(f, focus, extended, padding):
     pile.append(urwid.Columns(resp, dividechars=1))
     return urwid.Pile(pile)
 
+## common save body parts
+def save_body(path, master, state, content):
+    if not path:
+        return
+    state.last_saveload = path
+    path = os.path.expanduser(path)
+    try:
+        f = file(path, "wb")
+        f.write(str(content))
+        f.close()
+    except IOError, v:
+        master.statusbar.message(v.strerror)
+
+def ask_save_body(k, master, state, content):
+    if k == "y":
+        master.path_prompt(
+            "Save message content: ",
+            state.last_saveload,
+            save_body,
+            master,
+            state,
+            content,
+        )
+
+def which_body_save(k, master, state, flow):
+    if k == "q":
+        master.path_prompt(
+            "Save request content: ",
+            state.last_saveload,
+            save_body,
+            master,
+            state,
+            flow.request.get_decoded_content(),
+        )
+    elif k == "r":
+        if flow.response:
+            master.path_prompt(
+                "Save response content: ",
+                state.last_saveload,
+                save_body,
+                master,
+                state,
+                flow.response.get_decoded_content(),
+            )
+        else:
+            master.statusbar.message("Flow has no response")
+
+## common copy_message parts 
+def copy_message( k, master, state, message):
+    if not message:
+        # only response could be None
```

Comment:
I created this check because in flowlist you can press "g" (copy response) in a flow that has no response yet. I guess i can move the check to flowlist, but has to be done somewhere. (i do not remember if this case can be triggered from flowview).  Do you want me to change this?